sdk/mobile-center/src/test/java/com/microsoft/azure/mobile/utils/crypto/CryptoTest.java

```diff
@@ -0,0 +1,363 @@
+        when(mKeyStore.getEntry(argThat(new ArgumentMatcher<String>() {
+
+            @Override
+            public boolean matches(Object argument) {
+                return String.valueOf(argument).contains(CIPHER_AES);
+            }
+        }), any(KeyStore.ProtectionParameter.class))).thenReturn(aesKey);
+        when(mCryptoFactory.getKeyGenerator(anyString(), anyString())).thenReturn(mock(CryptoUtils.IKeyGenerator.class));
+        final byte[] mockInitVector = "IV".getBytes();
+        when(mCipher.getBlockSize()).thenReturn(mockInitVector.length);
+        when(mCipher.getIV()).thenReturn(mockInitVector);
+        when(mCipher.doFinal(any(byte[].class), anyInt(), anyInt())).thenAnswer(new Answer<byte[]>() {
+
+            @Override
+            public byte[] answer(InvocationOnMock invocation) throws Throwable {
+                byte[] input = (byte[]) invocation.getArguments()[0];
+                int offset = (int) invocation.getArguments()[1];
+                int length = (int) invocation.getArguments()[2];
+                byte[] data = new byte[length];
+                System.arraycopy(input, offset, data, 0, length);
+                return data;
+            }
+        });
+
+        /* Mock ciphers. */
+        when(mCryptoFactory.getCipher(anyString(), anyString())).thenReturn(mCipher);
+    }
+
+    @Test
+    public void initCryptoConstants() {
+        assertNotNull(new CryptoConstants());
+    }
+
+    @Test
+    public void nullData() {
+        CryptoUtils cryptoUtils = new CryptoUtils(mContext, mCryptoFactory, Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1);
+        assertNull(cryptoUtils.encrypt(null));
+        CryptoUtils.DecryptedData nullDecryptedData = cryptoUtils.decrypt(null);
+        assertNull(nullDecryptedData.getDecryptedData());
+        assertNull(nullDecryptedData.getNewEncryptedData());
+    }
+
+    private void verifyNoCrypto(int apiLevel) throws Exception {
+        CryptoUtils cryptoUtils = new CryptoUtils(mContext, mCryptoFactory, apiLevel);
+        String encrypted = cryptoUtils.encrypt("anything");
+        assertEquals("None" + ALGORITHM_DATA_SEPARATOR + "anything", encrypted);
+        CryptoUtils.DecryptedData decryptedData = cryptoUtils.decrypt(encrypted);
+        assertEquals("anything", decryptedData.getDecryptedData());
+        assertNull(decryptedData.getNewEncryptedData());
+    }
+
+    @Test
+    public void noCryptoInIceCreamSandwich() throws Exception {
+        verifyNoCrypto(Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1);
+        verifyStatic(never());
+        KeyStore.getInstance(anyString());
+    }
+
+    @Test
+    public void keyStoreNotFound() throws Exception {
+        when(KeyStore.getInstance(ANDROID_KEY_STORE)).thenThrow(new KeyStoreException());
+        verifyNoCrypto(Build.VERSION_CODES.KITKAT);
+        verifyStatic();
+        KeyStore.getInstance(anyString());
+    }
+
+    @Test
+    public void rsaFailsToLoadWhenPreferred() throws Exception {
```

Comment:
But is that a good convention anyway? Seem redundant to have verify in method names for a test class.